src/test/java/tw/joi/energy/domain/PricePlanTest.java

```
@@ -1,39 +1,46 @@
+    @DisplayName("Get energy supplier should return supplier if not null")
     public void get_energy_supplier_should_return_the_energy_supplier_given_supplier_is_existent() {
-        PricePlan pricePlan = new PricePlan(null, "Energy Supplier Name", null, null);
+        PricePlan pricePlan = new PricePlan("Test Plan Name", "Energy Supplier Name", BigDecimal.ONE, emptyList());
 
         assertThat(pricePlan.getEnergySupplier()).isEqualTo("Energy Supplier Name");
     }
 
     @Test
-    public void get_price_should_return_the_base_price_given_an_ordinary_date_time() throws Exception {
-        LocalDateTime normalDateTime = LocalDateTime.of(2017, Month.AUGUST, 31, 12, 0, 0);
-        PricePlan pricePlan = new PricePlan(null, null, BigDecimal.ONE, emptyList());
+    @DisplayName("Get price should return price given non-peak date and time")
+    public void get_price_should_return_the_base_price_given_an_ordinary_date_time() {
+        ZonedDateTime nonPeakDateTime = ZonedDateTime.of(LocalDateTime.of(2017, Month.AUGUST, 31, 12, 0, 0),
+            ZoneId.of("GMT"));
+        // the price plan has no peak days....
+        PricePlan pricePlan = new PricePlan("test plan", "test supplier", BigDecimal.ONE, emptyList());
 
-        BigDecimal price = pricePlan.getPrice(normalDateTime);
+        BigDecimal price = pricePlan.getPrice(nonPeakDateTime);
 
-        assertThat(price).isCloseTo(BigDecimal.ONE, Percentage.withPercentage(1));
+        assertThat(price).isEqualTo(BigDecimal.ONE);
     }
 
     @Test
+    @DisplayName("Get unit rate should return unit rate if no null")
     public void get_unit_rate_should_return_unit_rate_given_unit_rate_is_present() {
-        PricePlan pricePlan = new PricePlan(null, null, BigDecimal.TWO, null);
-        pricePlan.setPlanName("test-price-plan");
-        pricePlan.setEnergySupplier("test-energy-supplier");
+        PricePlan pricePlan = new PricePlan("test-price-plan", "test-energy-supplier", BigDecimal.TWO, emptyList());
```

This is obviously a good change, but I'll note that Tanuj's comment on the commit that set up the previous version was "add a layer of dust", in which he also make the fields non-final and added the setters.

I assume he was adding intentional smells.  We don't need to stick with that, but I'm curious if we're all aligned on exactly what smells, if any, we should have in there.

---

Lets discuss that internally. I did leave in some dubious decisions about throwing Exceptions rather than, say, returning Optional.